src/host/getset.cpp

```diff
@@ -357,6 +357,17 @@ void ApiRoutines::GetNumberOfConsoleMouseButtonsImpl(ULONG& buttons) noexcept
             WI_ClearFlag(gci.Flags, CONSOLE_USE_PRIVATE_FLAGS);
         }
 
+        if (WI_IsFlagSet(mode, ENABLE_MOUSE_INPUT) && !WI_IsFlagSet(context.InputMode, ENABLE_MOUSE_INPUT))
+        {
+            gci.GetActiveInputBuffer()->PassThroughEnableButtonEventMouseMode(true);
+            gci.GetActiveInputBuffer()->PassThroughEnableSGRExtendedMouseMode(true);
+        }
+        else if (!WI_IsFlagSet(mode, ENABLE_MOUSE_INPUT) && WI_IsFlagSet(context.InputMode, ENABLE_MOUSE_INPUT))
+        {
+            gci.GetActiveInputBuffer()->PassThroughEnableButtonEventMouseMode(false);
+            gci.GetActiveInputBuffer()->PassThroughEnableSGRExtendedMouseMode(false);

```

Comment:
I'd suggest that we just combine them all into one function. "PassThroughWin32MouseRequest" which enables or disables all requisite modes (since there's no reason to have all two or three in the input buffer interface)

Reply:
Yep makes sense